syncclient/client.py

```diff
@@ -43,18 +43,65 @@ def get_browserid_assertion(login, password, fxa_server_url=FXA_SERVER_URL,
     return bid_assertion, hexlify(sha256(keyB.encode('utf-8')).digest()[0:16])
 
 
+class SyncClientError(Exception):
+    """An error occured in SyncClient."""
+
+
+class TokenserverClient(object):
+    """Client for the Firefox Sync Token Server.
+    """
+    def __init__(self, bid_assertion, client_state,
+                 server_url=TOKENSERVER_URL):
+        self.bid_assertion = bid_assertion
+        self.client_state = client_state
+        self.server_url = server_url
+
+    def get_hawk_credentials(self):
+        """Asks for new temporary token given a BrowserID assertion"""
+        authorization = 'BrowserID %s' % encode_header(self.bid_assertion)
+        headers = {
+            'Authorization': authorization,
+            'X-Client-State': self.client_state
+        }
+        url = urlparse.urljoin(self.server_url, '/1.0/sync/1.5')
```

Comment:
I don't believe it is: it is possible to actually run a server at a specific location ".e.g." example.org/tokenserver and this would be broken by this urljoin.